google-beta/resource_service_networking_connection.go

```diff
@@ -0,0 +1,232 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/servicenetworking/v1beta"
+)
+
+func resourceServiceNetworkingConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServiceNetworkingConnectionCreate,
+		Read:   resourceServiceNetworkingConnectionRead,
+		Update: resourceServiceNetworkingConnectionUpdate,
+		Delete: resourceServiceNetworkingConnectionDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceServiceNetworkingConnectionImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"network": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			// NOTE(craigatgoogle): This field is weird, it's required to make the Insert/List calls, however it's defined
+			// in the API as an output field called "peering", which uses "-" as a delimeter instead of ".". To alleviate user
+			// confusion I've opted to model the gcloud CLI's approach, calling the field "service" and accepting the same
+			// format as the CLI with the "." delimiter.
+			// See: https://cloud.google.com/vpc/docs/configure-private-services-access#creating-connection
+			"service": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"reserved_peering_ranges": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceServiceNetworkingConnectionCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	network := d.Get("network").(string)
+	serviceNetworkingNetworkName, err := retrieveServiceNetworkingNetworkName(config, network)
+	if err != nil {
+		return fmt.Errorf("Failed to find Service Networking Connection, err: %s", err)
+	}
+
+	connection := &servicenetworking.Connection{
+		Network: serviceNetworkingNetworkName,
+	}
+
+	var reservedPeeringRanges []string
+	for _, reservedPeeringRange := range d.Get("reserved_peering_ranges").([]interface{}) {
+		reservedPeeringRanges = append(reservedPeeringRanges, reservedPeeringRange.(string))
+	}
+	connection.ReservedPeeringRanges = reservedPeeringRanges
+
+	parentService := formatParentService(d.Get("service").(string))
+	op, err := config.clientServiceNetworking.Services.Connections.Create(parentService, connection).Do()
+	if err != nil {
+		return err
+	}
+
+	if err := serviceNetworkingOperationWait(config, op, "Create Service Networking Connection"); err != nil {
+		return err
+	}
+
+	connectionId := &connectionId{
+		Network: network,
+		Service: d.Get("service").(string),
+	}
+
+	d.SetId(connectionId.Id())
+
+	if err := resourceServiceNetworkingConnectionRead(d, meta); err != nil {
+		d.SetId("")
```

Comment:
usually we just do `return resourceServiceNetworkingConnectionRead(d, meta)`- if there's something about this resource that requires doing it differently, can you add a comment explaining why?